analyses/molecular-subtyping-HGG/02-HGG-molecular-subtyping-subset-files.R

```diff
@@ -105,37 +128,41 @@ hgg_lesions_df <- hgg_lesions_df %>%
 #### Filter metadata -----------------------------------------------------------
 
 # Filter metadata for `High-grade glioma` and samples that should be classified
-# as High-grade glioma based on defining lesions and are stranded RNA-seq
-# samples.
+# as High-grade glioma based on defining lesions
 hgg_metadata_df <- metadata %>%
   dplyr::filter(
     disease_type_new == "High-grade glioma" |
       sample_id %in% hgg_lesions_df$sample_id,
-    experimental_strategy == "RNA-Seq",
-    RNA_library == "stranded"
+    sample_type == "Tumor",
+    composition == "Solid Tissue"
   )
 
-# Write to file
-readr::write_tsv(hgg_metadata_df, file.path(subset_dir, "hgg_histologies.tsv"))
-
 #### Filter expression data ----------------------------------------------------
 
-# Filter to HGG samples only -- we can use hgg_metadata_df because it is subset
-# to RNA-seq samples
-stranded_expression <- stranded_expression %>%
-  dplyr::select(hgg_metadata_df$Kids_First_Biospecimen_ID)
-
-# Log2 transformation
-norm_expression <- log2(stranded_expression + 1)
-
-# Scale does column centering, so we transpose first
-long_zscored_expression <- scale(t(norm_expression), 
-                                  center = TRUE,
-                                  scale = TRUE)
+filter_process_expression <- function(expression_mat) {
+  # Filter to HGG samples only -- we can use hgg_metadata_df because it is
+  # subset to RNA-seq samples
+  filtered_expression <- expression_mat %>%
+    dplyr::select(intersect(hgg_metadata_df$Kids_First_Biospecimen_ID,
+                            colnames(expression_mat)))
+
+  # Log2 transformation
+  log_expression <- log2(filtered_expression + 1)
+
+  # Scale does column centering, so we transpose first
+  long_zscored_expression <- scale(t(log_expression),
+                                   center = TRUE,
+                                   scale = TRUE)
+  return(long_zscored_expression)
+}
 
 # Save matrix with all genes to file for downstream plotting
-readr::write_rds(long_zscored_expression,
-                 file.path(subset_dir, "hgg_zscored_expression.RDS"))
+filter_process_expression(stranded_expression) %>%
+  readr::write_rds(file.path(subset_dir,
+                             "hgg_zscored_expression.stranded.RDS"))
+filter_process_expression(polya_expression) %>%
+  readr::write_rds(file.path(subset_dir,
+                             "hgg_zscored_expression.polya.RDS"))
```

Comment:
I like the idea of the two separate results tables to start. When you're putting together the notebook that looks at the expression, it might be good to include a density plot that looks at the distributions of each of the three genes of interest. I would expect them to look similar since we've already done the z-score step but it might make it clear that we should do a bit more digging or revisit the scaling.

Reply:
Okay sounds good, I'll do that.